### PR TITLE
exodus-android-app: Migrate to github actions

### PR DESCRIPTION
**Pull request by theimpulson**
This PR resolves https://github.com/Exodus-Privacy/exodus-android-app/issues/113

### CODE REVIEW

**Comment by pnu-s** [comment]
Hi @theimpulson and thanks for the PR.

Any idea why the new github action did not start for this PR?

**Comment by theimpulson** [comment]
> Hi @theimpulson and thanks for the PR.
> 
> Any idea why the new github action did not start for this PR?
> 
> The only difference between this file and the one from the [exodus repository](https://github.com/Exodus-Privacy/exodus) is that this part:
> 
> ```
>  on: [pull_request, push]
> ```
> 
> is replaced by:
> 
> ```
> on:
>   push:
>     branches:
>       - master
>     tags:
>       - v*
>   pull_request:
>     branches:
>       - master
>   workflow_dispatch:
> ```

@pnu-s I think it might be related to permissions. As per official docs:

> When a first-time contributor submits a pull request to a public repository, a maintainer with write access may need to approve running workflows on the pull request. 

Reference: https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

**Comment by pnu-s** [comment]
There is no such button, so It does not seem like it is blocked by a permission issue.

image

If I can recall correctly, when I added actions to the other repositories, I could test them before merging (but I'm not 100% sure).

File: .github/workflows/main.yml
Comment (pnu-s) [inline .github/workflows/main.yml:1]:
```suggestion
---
name: Build

```

File: .github/workflows/main.yml
Comment (pnu-s) [inline .github/workflows/main.yml:2]:
```suggestion
on:
  push:
    branches:
      - master
    tags:
      - v*
  pull_request:
    branches:
      - master
  workflow_dispatch:
```

File: .github/workflows/main.yml
Comment (pnu-s) [inline .github/workflows/main.yml:12]:
```suggestion
        run: ./gradlew test

```

**Comment by theimpulson** [comment]
It does seem to be related to permissions. I tried opening a PR in my fork and that did trigger the action in it. Maybe the other repositories triggered action as you had perms/access to them.

Ref: https://github.com/theimpulson/exodus-android-app/pull/1

**Review by pnu-s — CHANGES_REQUESTED** [review]
For consistency purposes, made a couple of suggestions

File: .github/workflows/main.yml
Comment (theimpulson) [reply .github/workflows/main.yml:2]: Done

File: .github/workflows/main.yml
Comment (theimpulson) [reply .github/workflows/main.yml]: Done

File: .github/workflows/main.yml
Comment (theimpulson) [reply .github/workflows/main.yml]: Done

**Review by pnu-s — APPROVED** [review]

**Comment by pnu-s** [comment]
@theimpulson I tried making a change in the Exodus Privacy organization settings, could you please try to push a new change (or just force-push the same changes) to make sure GitHub is taking these changes into account?

If this does not work, I'm running out of ideas! :confounded:

**Comment by theimpulson** [comment]
> @theimpulson I tried making a change in the Exodus Privacy organization settings, could you please try to push a new change (or just force-push the same changes) to make sure GitHub is taking these changes into account?
> 
> If this does not work, I'm running out of ideas! 😖

@pnu-s doesn't seem like that worked. I think that:

- You can create a new test branch on this repo, cherry-pick this commit and open a test PR to see if that triggers the build (which I think will), or/and
- Merge this MR and see if build gets triggered on master for `push` and I will fpush my other MR once to see if the build gets triggered for that `push`

**Comment by pnu-s** [comment]
Tested it in https://github.com/Exodus-Privacy/exodus-android-app/pull/116 and it works indeed as expected.

Unsure why it does not start here, maybe because it comes from a fork but I allowed this in our settings.

Anyway, will merge this PR, thanks again @theimpulson :tada:

**Pull request merged**
